scipy/spatial/transform/rotation.py

```diff
@@ -1,12 +1,71 @@
 from __future__ import division, print_function, absolute_import
 
 import numpy as np
+import scipy.linalg
+import warnings
 
 class Rotation(object):
     """Rotation in 3 dimensions.
 
     This class will include initializers from different representations, converters
     and some useful algorithms such as quaternion slerp and rotation estimation.
+
+    For initializing Rotations usage of `from_...` methods such as `from_quaternion`
+    is recommended instead of using `__init__`.
+
+    Methods
+    -------
+    from_quaternion
     """
-    def __init__(self):
-        pass
+    def __init__(self, quat, normalized=False):
```

Comment:
One other thing we could consider is supporting 3-element `(x,y,z)` quaternions if `normalized=True`, because `w` can be computed in this case.

Reply:
It is a bit tricky: you can't determine the sign of the last element, which is crucial. So we have to pass 4 numbers always, even though we know their sum of squares is 1.

Reply:
We can say that we will always use the positive variant of the value. If people want something else they can provide it.

Reply:
Right, `w=1` and `w=-1` both represent the same overall rotation, but `w=-1` represents a rotation "the longer way around". While not important right now, this will have a major effect when we implement interpolation algorithms later on.

We could do a default of `w=1` or something, along with appropriate warnings, thoughts on that?

Reply:
To be honest, I think it is very unconventional to represent rotation quaternions as 3 numbers. I never saw them used that way or implemented that way. I think it will be confusing to allow such representation, I just don't see where it can occur.

Reply:
I have used a program that had this convention, but if you haven't seen it at all then it's probably not widely used enough to do here. We can always add it later, too, if it turns out others also have seen it.

Reply:
Interesting. I'm pretty sure it's not widely used.

Let me also explain, why I think it is tricky and (sort of) bug prone. If we have 4 number `(x, y, z, w)` then in order to convert them to 3 numbers with a convention that `w > 0` we first need to check the sign of `w` and if it's negative --- negate `x, y, z`. If we forget to do that and just throw away `w`, then we will get a completely different rotation. And considering that all the algorithms with quaternions are conveniently expressed with `(x, y, z, w)` it seems indeed better to just keep `(x, y, z, w)` and don't use the sign tricks. In other words `(x, y, z, w)` representation is consistent and transparent, whereas `(x, y, z)` relies on an assumption, which is not really a part of quaternion mathematics.

And what Aditya said about interpolation --- this is quite interesting view as well. I didn't think about it and it is not always applicable, i.e. it is often said that `q` and `-q` are the same rotations. But looks like in fact `q` and `-q` might be viewed differently for interpolation problems, which is great.

So I suggest not to support such representation for now. Perhaps ask on the mailing list when everything will be more ready.